web/lib/util/embed.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react'
+
+export function inIframe() {
+  try {
+    return window.self !== window.top
+  } catch (e) {
+    return true
+  }
+}
+
+// use a hook so this calculation happens client side
+export function useInIframe() {
```

Comment:
I tend to like verbose names, but to me `useIsInIframe` is better. Wait! Just do `useIsIframe`. I think that's best.